op-challenger/game/fault/types/position.go

```diff
@@ -25,6 +25,15 @@ func (p Position) MoveRight() Position {
 	}
 }
 
+// Localize returns a new position for a subtree.
+// reduced is the number of levels to reduce the depth by.
+func (p Position) Localize(reduced uint64) Position {
+	newPosDepth := uint64(p.depth) - reduced
+	nodesAtDepth := 1 << newPosDepth
+	newIndexAtDepth := p.indexAtDepth % nodesAtDepth
+	return NewPosition(int(newPosDepth), newIndexAtDepth)
+}
```

Comment:
We should add a bunch of unit tests around this. It will be very hard to debug if it's wrong (and I'm old enough to have learnt not to trust my abilities with this kind of math... 😄).

I'm not really sold on the `Localize` name either but hard to think of something better.  The long version would be `RelativeToAncestorAtDepth` which I'd be seriously tempted to go with because it's quite explicit about what its doing.

Reply:
I'm ok with `RelativeToAncestorAtDepth` as well. If methods have good inline documentation, it's propagated to tooltips and descriptors so it will be understandable. In fact, even having a method with a verbose name like `RelativeToAncestorAtDepth` can be very confusing without good inline docs IMHO. It's a habit I think our team needs to work on across the codebase.